cmd/gokr-packer/packer.go

```diff
@@ -298,6 +294,93 @@ func findEnvFiles() (map[string]string, error) {
 	return contents, nil
 }
 
+func addToFileInfo(parent *fileInfo, path string) (error, time.Time) {
+	entries, err := os.ReadDir(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, time.Time{}
+		}
+		return err, time.Time{}
+	}
+
+	var latestTime time.Time
+	for _, entry := range entries {
+		// get existing file info
+		var fi *fileInfo
+		for _, ent := range parent.dirents {
+			if ent.filename == entry.Name() {
+				fi = ent
+				break
+			}
+		}
+
+		info, err := entry.Info()
+		if err != nil {
+			return err, time.Time{}
+		}
+
+		if latestTime.Before(info.ModTime()) {
+			latestTime = info.ModTime()
+		}
+
+		// or create if not exist
+		if fi == nil {
+			fi = &fileInfo{
+				filename: entry.Name(),
+				mode:     info.Mode(),
+			}
+			parent.dirents = append(parent.dirents, fi)
+		} else {
+			// file overwrite is not supported -> return error
+			if !entry.IsDir() || fi.fromHost != "" || fi.fromLiteral != "" {
+				return fmt.Errorf("file alreayd exist in rootfs: %s", filepath.Join(path, entry.Name())), time.Time{}
+			}
+		}
+
+		// add content
+		if entry.IsDir() {
+			err, modTime := addToFileInfo(fi, filepath.Join(path, entry.Name()))
+			if err != nil {
+				return err, time.Time{}
+			}
+			if latestTime.Before(modTime) {
+				latestTime = modTime
+			}
+		} else {
+			fi.fromHost = filepath.Join(path, entry.Name())
+		}
+	}
+
+	return nil, latestTime
+}
+
+func findExtraFiles() (map[string]*fileInfo, error) {
+	buildPackages := buildPackagesFromFlags()
+	extraFiles := make(map[string]*fileInfo, len(buildPackages))
+	for pkg := range buildPackages {
+		path := filepath.Join("extrafiles", pkg)
+
+		fi := new(fileInfo)
+		err, latestModTime := addToFileInfo(fi, path)
+		if err != nil {
+			return nil, err
+		}
+		if len(fi.dirents) == 0 {
+			continue
+		}
+
+		packageConfigFiles[pkg] = append(packageConfigFiles[pkg], packageConfigFile{
+			kind:         "started with extra files",
```

Comment:
Maybe instead of "be started with extra files", how about "include extra files in the root file system"? (You’ll need to move the "be" into the kind field.

Reply:
changed